### PR TITLE
Refactored staff_manage.php and student_manage.php tables

### PR DESCRIPTION
**Pull request by powertashton**

### CODE REVIEW

File: IB Diploma/staff_manage.php [hunk 1]
```diff
@@ -31,82 +33,26 @@
...
-
-    //Set pagination variable
-    $pagination = $_GET['page'] ?? 1;
-    if ((!is_numeric($pagination)) or $pagination < 1) {
-        $pagination = 1;
-    }
-
-    //SELECT NAMED
+    
     try {
```
Comment (SKuipers) [inline IB Diploma/staff_manage.php]:
```suggestion

```

File: IB Diploma/staff_manage.php [hunk 2]
```diff
@@ -31,82 +33,26 @@
...
-
-    if ($result->rowCount() < 1) { $page->addError(__('There are no staff to display.'));
-    } else {
-        if ($result->rowCount() > $_SESSION[$guid]['pagination']) {
-            printPagination($guid, $result->rowCount(), $pagination, $_SESSION[$guid]['pagination'], 'top');
-        }
+    $data = array();
+    $sql = "SELECT ibDiplomaCASStaffID, ibDiplomaCASStaff.role, surname, preferredName FROM ibDiplomaCASStaff JOIN gibbonPerson ON (ibDiplomaCASStaff.gibbonPersonID=gibbonPerson.gibbonPersonID) WHERE status='Full' ORDER BY role, surname, preferredName";
+    $result = $pdo->select($sql, $data)->toDataSet();
+    } catch (PDOException $e) { $page->addError($e->getMessage());}
```
Comment (SKuipers) [inline IB Diploma/staff_manage.php]:
```suggestion

```

File: IB Diploma/student_manage.php [hunk 3]
```diff
@@ -36,112 +39,28 @@
...
         returnProcess($guid, $_GET['return'], null, null);
     }
-
-    //Set pagination variable
-    $pagination = $_GET['page'] ?? 1;
-    if ((!is_numeric($pagination)) or $pagination < 1) {
-        $pagination = 1;
-    }
-
     try {
```
Comment (SKuipers) [inline IB Diploma/student_manage.php]:
```suggestion

```

File: IB Diploma/student_manage.php [hunk 4]
```diff
@@ -36,112 +39,28 @@
...
+        $result = $pdo->select($sql, $data)->toDataSet();
+    } catch (PDOException $e) { $page->addError(__('Students cannot be displayed.')); }
```
Comment (SKuipers) [inline IB Diploma/student_manage.php]:
```suggestion
```

**Review by SKuipers — CHANGES_REQUESTED** [review]
This is looking great! As a note, the try/catch blocks are no longer necessary in queries that use the $pdo or gateway classes, the error handling is built-in. I've highlighted a couple here so that you can automatically remove them through the GitHub UI. Thanks!

**Comment by powertashton** [comment]
> This is looking great! As a note, the try/catch blocks are no longer necessary in queries that use the $pdo or gateway classes, the error handling is built-in. I've highlighted a couple here so that you can automatically remove them through the GitHub UI. Thanks!

Have noted this down for future reference and made the changes for this pull request... I totally didn't completely forget this pull request existed for two months

**Review by SKuipers — APPROVED** [review]
Wow, somehow this PR dropped off my radar too 😅 Going to merge and test it out now. Thanks!

**Pull request merged**